src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,244 @@
+      setIsLoading(false);
+    }
+  };
+
+  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!nameValue || !emailValue || !commentValue) {
+      setHasInputValue(() => ({
+        hasName: !!nameValue,
+        hasEmail: !!emailValue.length,
+        hasComment: !!commentValue.length,
+      }));
+
+      return;
+    }
+
+    addNewComment();
+  };
+
+  const handleClearForm = () => {
+    setNameValue('');
+    setEmailValue('');
+    setCommentValue('');
+    setHasInputValue(fieldStatus);
+  };
+
+  return (
+    <form data-cy="NewCommentForm" onSubmit={handleSubmitForm}>
+      <div className="field" data-cy="NameField">
+        <label className="label" htmlFor="comment-author-name">
+          Author Name
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
+          <input
+            type="text"
+            name="name"
+            id="comment-author-name"
+            placeholder="Name Surname"
+            className={classNames(
+              'input',
+              { 'is-danger': !hasInputValue.hasName },
+            )}
+            value={nameValue}
+            onChange={handleInputChange}
+          />
+
+          <span className="icon is-small is-left">
+            <i className="fas fa-user" />
+          </span>
+          {!hasInputValue.hasName && (
+            <span
+              className="icon is-small is-right has-text-danger"
+              data-cy="ErrorIcon"
+            >
+              <i className="fas fa-exclamation-triangle" />
+            </span>
+          )}
+        </div>
+        {!hasInputValue.hasName && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Name is required
+          </p>
+        )}
+
+      </div>
+
+      <div className="field" data-cy="EmailField">
+        <label className="label" htmlFor="comment-author-email">
+          Author Email
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
+          <input
+            type="text"
+            name="email"
+            id="comment-author-email"
+            placeholder="email@test.com"
+            className={classNames(
+              'input',
+              { 'is-danger': !hasInputValue.hasEmail },
+            )}
+            value={emailValue}
+            onChange={handleInputChange}
+          />
+
+          <span className="icon is-small is-left">
+            <i className="fas fa-envelope" />
+          </span>
+          {!hasInputValue.hasEmail && (
+            <span
+              className="icon is-small is-right has-text-danger"
+              data-cy="ErrorIcon"
+            >
+              <i className="fas fa-exclamation-triangle" />
+            </span>
+          )}
+        </div>
+        {!hasInputValue.hasEmail && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Email is required
+          </p>
+        )}
```

Comment:
move into separated component and reuse throughout your code

Reply:
I moved it to a separate component, but I don't want to put it into a separate file. Also, there is a div in the middle, so I had to make 2 components